src/app/modules/analytics-config/components/import-widget-dialog/import-widget-dialog.component.ts

```diff
@@ -0,0 +1,100 @@
+You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing,
+software distributed under the License is distributed on an
+"AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+KIND, either express or implied.  See the License for the
+specific language governing permissions and limitations
+under the License.
+*/
+import { Component, OnInit, ChangeDetectorRef } from '@angular/core';
+import { PersistenceService } from 'arlas-wui-toolkit/services/persistence/persistence.service';
+import { ZONE_WUI_BUILDER } from '@services/startup/startup.service';
+import { DataResource, DataWithLinks } from 'arlas-persistence-api';
+import { FormGroup, FormControl, Validators } from '@angular/forms';
+import { MatSelectChange, MatCheckboxChange } from '@angular/material';
+import { Config, AnalyticConfig, ContributorConfig, AnalyticComponentConfig } from '@services/main-form-manager/models-config';
+import { Layer, MapConfig } from '@services/main-form-manager/models-map-config';
+import { MainFormService } from '@services/main-form/main-form.service';
+
+@Component({
+  selector: 'app-import-widget-dialog',
+  templateUrl: './import-widget-dialog.component.html',
+  styleUrls: ['./import-widget-dialog.component.scss']
+})
+export class ImportWidgetDialogComponent implements OnInit {
+
+  public configs: DataWithLinks[];
+  public dashboardConfigJson: Config;
+  public importWidgetFormGroup: FormGroup;
+  public analytics =  new Map<string, Array<AnalyticConfig>>();
+  public contributors =  new Map<string, string>();
+
+  public selectedWidgets: Array<AnalyticComponentConfig> = new Array();
+  public selectedWidgetsSet: Set<AnalyticComponentConfig> = new Set();
+
+  constructor(
+    private persistenceService: PersistenceService,
+    private mainformService: MainFormService,
+    private cdr: ChangeDetectorRef
+  ) { }
+
+  public ngOnInit() {
+    this.importWidgetFormGroup = new FormGroup(
+      {
+        dashboard: new FormControl(null, Validators.required),
+        componentConfig: new FormControl(null, Validators.required)
+
+      }
+    );
+
+    if (this.persistenceService.isAvailable) {
+      this.persistenceService.list(ZONE_WUI_BUILDER, null, null, 'desc').subscribe({
+        next: (data: DataResource) => {
+          this.configs = data.data.filter(
+            dash => (JSON.parse(dash.doc_value) as Config).arlas.server.collection.name === this.mainformService.getCollections()[0]
+          );
+        }
+      });
+    }
+  }
+
+  public onChange(event) {
+    if (event.checked) {
+      this.selectedWidgetsSet.add(event.source.value);
+    } else {
+      this.selectedWidgetsSet.delete(event.source.value);
+    }
+    this.selectedWidgets = Array.from(this.selectedWidgetsSet);
+    this.cdr.detectChanges();
+  }
+
+  public getWidgets(event: MatSelectChange) {
+    this.cdr.detectChanges();
```

Comment:
Why `detectChanges` at start of an event ?

Reply:
It's a misattention line
I ll put it at the end

Reply:
fixed